src/Agent.Worker/ContainerOperationProvider.cs

```diff
@@ -106,6 +98,27 @@ public async Task StartContainerAsync(IExecutionContext executionContext, object
                 throw new NotSupportedException(StringUtil.Loc("MinRequiredDockerClientVersion", requiredDockerVersion, _dockerManger.DockerPath, dockerVersion.ClientVersion));
             }
 
+            executionContext.Debug($"Delete stale container networks from previous jobs");
+            int networkPruneExitCode = await _dockerManger.DockerNetworkPrune(executionContext);
+            if (networkPruneExitCode != 0)
+            {
+                executionContext.Warning($"Delete stale container networks failed, docker network prune fail with exit code {networkPruneExitCode}");
+            }
+
+            // Create local docker network for this job to avoid port conflict when multiple agents run on same machine.
+            // All containers within a job join the same network
+            var existingNetwork = executionContext.Variables.Get(Constants.Variables.Agent.ContainerNetwork);
```

Comment:
what's the case `Constants.Variables.Agent.ContainerNetwork` is already set at this point? user set it in their definition?

Reply:
Fixed. This is supported